File: 8-StringToIntegerAtoi/8-StringToIntegerAtoi.py
# Last updated: 07/04/2026, 11:43:44
class Solution:
    def myAtoi(self, s: str) -> int:

        ans = i =  0 
        sign = 1 

        s = s.lstrip()
        if not s :
            return ans

        if s[0] == "-" :
            sign = -1
            i+=1
        elif s[0] == "+":
            i+=1

        
        while i < len(s):
            if not s[i].isdigit():
                break
            else:
                ans = (ans *10) + int(s[i])
            i+=1

        ans = ans *sign
        if ans > 2**31-1:
            return 2**31-1
        elif ans <= -2**31 :
            return -2**31
        else:
            return ans

        


        

File: 8-StringToIntegerAtoi/8-StringToIntegerAtoi.cpp
// Last updated: 07/04/2026, 11:43:38
class Solution {
public:
    int myAtoi(string s) {
        int i = 0, n = s.size();
        int sign = 1;
        long ans = 0;  // use long to detect overflow

        // skip spaces
        while (i < n && s[i] == ' ') i++;

        // sign
        if (i < n && (s[i] == '+' || s[i] == '-')) {
            sign = (s[i] == '-') ? -1 : 1;
            i++;
        }

        // digits
        while (i < n && isdigit(s[i])) {
            ans = ans * 10 + (s[i] - '0');

            // overflow check
            if (sign * ans > INT_MAX) return INT_MAX;
            if (sign * ans < INT_MIN) return INT_MIN;

            i++;
        }

        return (int)(sign * ans);
    }
};